### PR TITLE
[google_sign_in_web] Fixes force unwrap on values that can be null

### PR DESCRIPTION
**Pull request by kamilpowalowski**
During Google Sign-in, the code uses two force unwraps on values (name and picture) that can be not present in the response. This cause an unhandled error that blocks sign-in.

Fixes https://github.com/flutter/flutter/issues/130002 reported by me. The bug report describes how to get that error together with a screenshot of a given line.

My PR fixes that and add additional test for the future.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/google_sign_in/google_sign_in_web/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.12.1
```
Comment (stuartmorgan) [inline packages/google_sign_in/google_sign_in_web/CHANGELOG.md:1]: This is just a bugfix, not an API change, so it should be 0.12.0+3

**Review by stuartmorgan — APPROVED** [review]
Thanks for the contribution! LGTM other than the version.

@ditman for secondary review.

File: packages/google_sign_in/google_sign_in_web/CHANGELOG.md (hunk 1 above)
Comment (kamilpowalowski) [reply packages/google_sign_in/google_sign_in_web/CHANGELOG.md]: Thank you, updated!

File: packages/google_sign_in/google_sign_in_web/example/integration_test/src/jwt_examples.dart [hunk 2]
```diff
@@ -38,6 +44,22 @@ const String goodJwtToken =
...
     'eyJlbWFpbCI6ImFkdWx0bWFuQGV4YW1wbGUuY29tIiwic3ViIjoiMTIzNDU2IiwibmFtZSI6IlZpbmNlbnQgQWR1bHRtYW4iLCJwaWN0dXJlIjoiaHR0cHM6Ly90aGlzcGVyc29uZG9lc25vdGV4aXN0LmNvbS9pbWFnZT94PS5qcGcifQ';
 
+/// A JWT token with minimal set of predefined values.
+///
+/// 'email': 'adultman@example.com',
+/// 'sub': '123456'
+///
+/// Signed with HS256 and the private key: 'symmetric-encryption-is-weak'
+const String minimalJwtToken =
+    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.$minimalPayload.UTAe7dpdtFIMwsOqkZkjyjqyHnho5xHCcQylUFmOutM';
```
Comment (ditman) [inline packages/google_sign_in/google_sign_in_web/example/integration_test/src/jwt_examples.dart:54]: Thanks for adding a test!

I'm glad you managed to create one of these, it's not a trivial process!

File: packages/google_sign_in/google_sign_in_web/lib/src/utils.dart [hunk 3]
```diff
@@ -72,8 +72,8 @@ GoogleSignInUserData? gisResponsesToUserData(
   return GoogleSignInUserData(
     email: payload['email']! as String,
     id: payload['sub']! as String,
-    displayName: payload['name']! as String,
-    photoUrl: payload['picture']! as String,
+    displayName: payload['name'] as String?,
+    photoUrl: payload['picture'] as String?,
```
Comment (ditman) [inline packages/google_sign_in/google_sign_in_web/lib/src/utils.dart:76]: This makes total sense. Even the documentation of `GoogleSignInUserData` has the following comment on top of `displayName` and `photoUrl`:

```
/// Not guaranteed to be present for all users, even when configured.
```

**Review by ditman — APPROVED** [review]
Thanks for noticing this, and following the whole contribution procedure!

LGTM!

**Pull request merged**